ppl/archive/import.go

```diff
@@ -53,34 +56,53 @@ func Import(ctx context.Context, ark *Archive, zctx *resolver.Context, r zbuf.Re
 // provided read stream is closed. Add some kind of timeout functionality that
 // periodically flushes stale tsDirWriters.
 type Writer struct {
-	ark     *Archive
-	ctx     context.Context
-	writers map[tsDir]*tsDirWriter
+	ark      *Archive
+	cancel   context.CancelFunc
+	ctx      context.Context
+	errgroup *errgroup.Group
+	mu       sync.RWMutex
```

Comment:
Maybe call this writersMu since it only protects the writers field?

Reply:
It looks like you've got `mu` trying to just guard the `writers` map, but I think there are code paths where more locking is needed. `memBuffered` for example is incremented I think in both the Write() call and the flush path.

I suggest changing `mu` to a `sync.Mutex`, and calling lock/defer-unlock on it at the top of: `Write`, `Close`, and `flushStaleWriters`'. If you do that, I think you don't need `modts` to be an atomic type.

Reply:
@alfred-landrum good call on memBuffered. WIll fix.

Reply:
@alfred-landrum I was concerned about doing too much mutexing here because this is in the critical path for import. That said perhaps doing as you suggest has a negligible impact.

Reply:
@alfred-landrum I went ahead and followed @nwt 's suggestion. I couldn't get away from not wanting to put a mutex over the whole thing, worried this would slow things down. For myself, I also easier find I'm less worried about deadlocks if I only have to worry about a mutex on the writer (I followed @nwt's suggestion to name it `writersMu`).

I used an atomic to make memBuffered thread safe.